table/block_based/filter_policy.cc

```diff
@@ -207,12 +215,47 @@ class FastLocalBloomBitsBuilder : public XXPH3FilterBitsBuilder {
   ~FastLocalBloomBitsBuilder() override {}
 
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) override {
+    return Finish(buf, nullptr);
+  }
+  // REQUIRED: if cache_res_mgr_ != nullptr, then buf_cache_res_handle !=
+  // nullptr
```

Comment:
Note to @pdillinger:

I think it makes sense to have this assertion. Otherwise we might end up in situation where there is actual memory reservation but it will be released way earlier than the final filter get de-allocated since there is no handle to rely that reservation.

This PR also makes sure all path calling here with `cache_res_mgr_ != nullptr` (i.e, only `BlockBasedTableBuilder` and `FilterBench`) provide a non-null handle.

Reply:
No longer needed due to API change